src/carma-platform/guidance_scripts.py

```diff
@@ -993,6 +997,80 @@ def run_guidance_steering_analysis(
     return (is_passed, stats, plt.gcf(), error_angles, common_timestamps)
 
 
+def extract_steering_data(mcap_path, start_time=None, end_time=None):
+    """
+    Extracts steering data from MCAP file using only steering report topics.
+    Returns the extracted data and used topic for the first valid topic found.
+    Supports PACMOD, DATASPEED, and NEWEAGLE steering report topics.
+
+    Args:
+        mcap_path: Path to MCAP file
+        start_time: Time to start the extraction
+        end_time: Time to end the extraction
+
+    Returns:
```

Comment:
can you clarify what is in the extracted data?

Reply:
Added note on what is being returned